v2.3.0a2 regression: RuntimeWarning on free-threading builds

I got this from the mpmath CI:
```
=================================== FAILURES ===================================
_____________________________ test_mpmath_version ______________________________
[gw2] linux -- Python 3.14.0 /opt/hostedtoolcache/Python/3.14.0/x64-freethreaded/bin/python

    def test_mpmath_version():
        c = Console(f'{sys.executable} -m mpmath --version')
    
        assert c.expect(pexpect.EOF) == 0
>       assert c.before.startswith('1.')
E       assert False
E        +  where False = <built-in method startswith of str object at 0x2b348998a40>('1.')
E        +    where <built-in method startswith of str object at 0x2b348998a40> = "<frozen importlib._bootstrap>:491: RuntimeWarning: The global interpreter lock (GIL) has been enabled to load module ...ehavior and keep the GIL disabled (at your own risk), run with PYTHON_GIL=0 or -Xgil=0.\r\n1.4.0b3.dev6+gf444dac15\r\n".startswith
E        +      where "<frozen importlib._bootstrap>:491: RuntimeWarning: The global interpreter lock (GIL) has been enabled to load module ...ehavior and keep the GIL disabled (at your own risk), run with PYTHON_GIL=0 or -Xgil=0.\r\n1.4.0b3.dev6+gf444dac15\r\n" = <test_cli.Console object at 0x2b344183490>.before

mpmath/tests/test_cli.py:152: AssertionError
=============================== warnings summary ===============================
<frozen importlib._bootstrap>:491
<frozen importlib._bootstrap>:491
<frozen importlib._bootstrap>:491
<frozen importlib._bootstrap>:491
<frozen importlib._bootstrap>:491
  <frozen importlib._bootstrap>:491: RuntimeWarning: The global interpreter lock (GIL) has been enabled to load module 'gmpy2.gmpy2', which has not declared that it can run safely without the GIL. To override this behavior and keep the GIL disabled (at your own risk), run with PYTHON_GIL=0 or -Xgil=0.

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```
See e.g. https://github.com/mpmath/mpmath/actions/runs/19843235944/job/56856376550

Pinning to v2.3.0a1 fixes the issue.

--- a/src/gmpy2.c
+++ b/src/gmpy2.c
@@ -944,11 +944,7 @@ static PyModuleDef_Slot Pygmpy_slots[] = {
      Py_MOD_MULTIPLE_INTERPRETERS_NOT_SUPPORTED},
 #  endif
 #  if PY_VERSION_HEX >= 0x030D0000
-#    ifdef MPFR_USE_THREAD_SAFE
-    {Py_mod_gil, Py_MOD_GIL_NOT_USED},
-#    else
     {Py_mod_gil, Py_MOD_GIL_USED},
-#    endif
 #  endif
     {0, NULL}
 };
@@ -969,5 +965,15 @@ static struct PyModuleDef gmpy_module = {
 
 PyMODINIT_FUNC PyInit_gmpy2(void)
 {
+#if PY_VERSION_HEX >= 0x030D0000
+    if (mpfr_buildopt_tls_p()) {
+        for (PyModuleDef_Slot *slot = Pygmpy_slots; slot->slot; slot++) {
+            if (slot->slot == Py_mod_gil) {
+                slot->value = Py_MOD_GIL_NOT_USED;
+                break;
+            }
+        }
+    }
+#endif
     return PyModuleDef_Init(&gmpy_module);
 }

